iati/tests/test_default.py

```diff
@@ -88,18 +93,36 @@ def test_default_codelist_invalid_at_all_versions(self, name, standard_version_o
         assert 'There is no default Codelist in version' in str(excinfo.value)
 
     def test_default_codelists_type(self, codelist_lengths_by_version):
-        """Check that the default Codelists are of the correct type.
-
-        Todo:
-            Check internal values beyond the codelists being the correct type.
-        """
+        """Check that the default Codelists are of the correct type."""
         codelists = iati.default.codelists(codelist_lengths_by_version.version)
 
         assert isinstance(codelists, dict)
         assert len(codelists.values()) == codelist_lengths_by_version.expected_length
         for codelist in codelists.values():
             assert isinstance(codelist, iati.Codelist)
 
+    def test_default_codelists_code_values(self, standard_version_optional, codelists_with_no_name_codes):
+        """Check that the default Codelists have Codes with relevant data in them."""
+        codelists = iati.default.codelists(*standard_version_optional)
+        relevant_codelists = [codelist for codelist in codelists.values() if codelist.name not in codelists_with_no_name_codes]
+
+        assert isinstance(codelists, dict)
+        for codelist in relevant_codelists:
+            for code in codelist.codes:
+                assert isinstance(code, iati.Code)
+                assert code.name != ''
+
+    def test_default_codelists_no_name_codes_have_no_name(self, standard_version_optional, codelists_with_no_name_codes):
+        """Check that Codelists with Codes that supposedly have no name do have no name."""
```

Comment:
This should be reworded for clarity.

Reply:
https://github.com/IATI/pyIATI/pull/214/commits/ebdf72bb1c0f67277e5144377601b59ed9c6076d